### PR TITLE
proposal: CPC-CHARTER to Stage 3

### PR DESCRIPTION
**Pull request by MylesBorins**
To be brought to the board for final approval

### CODE REVIEW

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+## Section 1. Guiding Principle.
+
+The OpenJS Foundation will operate transparently, openly,
+collaboratively, and ethically. Project proposals, timelines, and status
+must not merely be open, but also easily visible to outsiders.
+
+## Section 2. Evolution of OpenJS Foundation Governance.
+
+Most large, complex open source communities have both a business and a
+technical governance model. OpenJS Foundation’s technical leadership
```
Comment (Trott) [inline proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]: This first sentence of this paragraph should be omitted IMO.

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+### Observers
+
+Observers are free to attend meetings and participate in the work of the
+CPC as well as the consensus seeking process.
+
+### Regular members
+
+Regular members have made a commitment to be involved in an ongoing manner
+and take on roles and responsibilities as outlined below.
+Regular member is implied when memership type is not
```
Comment (Trott) [inline proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]:
```suggestion
Regular member is implied when membership type is not
```

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+
+### Voting members
+
+Voting members are selected as follows:
+
+* Each Impact project may nominate up to two members through a process
+of their choosing. Once nominated the member must be ratified by the
+CPC Voting members before becoming a Voting member.
+
+* up to two Voting members may be nominated by the non Impact projects based
```
Comment (Trott) [inline proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]:
```suggestion
* up to two Voting members may be nominated by the non-Impact projects based
```

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+Section 5. Responsibilities and Expectations of the CPC
+
+Subject to such policies as may be set by the Board, the CPC is responsible for
+ensuring: 
+
+   1. When needed, members of the CPC will be expected to create subcommittees
+      or working groups. The CPC will delegate responsibilities and empower
+      these groups to make decisions. Any of the responsibilities listed below
+      not identified as being responsibilities of the Voting members may be delegated.
+      For the remanining responsibilities, day-to-day work, investigation, and building
```
Comment (Trott) [inline proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]:
```suggestion
      For the remaining responsibilities, day-to-day work, investigation, and building
```

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+      recommentations may be delegated, however, the final responsibility will
```
Comment (Trott) [inline proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]:
```suggestion
      recommendations may be delegated, however, the final responsibility will
```

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+   1. Defining common practices to be implemented across OpenJS Foundation projects, if any.
+   1. Mediating technical conflicts between OpenJS Foundation Projects when attempts to resolve
+      those conflicts within the Project were unsuccessful. 
+   1. Managing the process for creating a yearly budget and approving a yearly budget to be
+      presented to the board.
+   1. The CPC serves as the OpenJS Foundation's primary technical
+      liaison body with external open source projects, consortiums and groups and is
+      also responsible for ensuring that the OpenJS Foundation has appropriate technical
+      participation in standards bodies by finding an encouraging members from
+      individual projects to particiapte in these bodies.
```
Comment (Trott) [inline proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]:
```suggestion
      individual projects to participate in these bodies.
```

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+   1. Guiding projects into appropriate Foundation paths.
+   1. Shared resources, tooling and funding for projects and Commons.js (Events
+      Collaborator Summit, Travel fund).
+   1. Supporting projects in their enforcement of Code of Conduct
+      violations and escalation.
+   1. Developing a framework for community end-user feedback.
+   1. Managing programs to offer mentorship to external individuals.
+   1. Managing developer outreach programs.
+   1. Supporting project advocacy and recognition programs.
+   1. Identifing, recruiting and engaging prospective projects.
```
Comment (Trott) [inline proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]:
```suggestion
   1. Identifying, recruiting and engaging prospective projects.
```

**Comment by Trott** [comment]
All instances of `decision making` in this doc would be better hyphenated. (Let me know if I should just go to town editing these small things myself.)

**Comment by Trott** [comment]
> All instances of `decision making` in this doc would be better hyphenated. (Let me know if I should just go to town editing these small things myself.)

Er, actually, I don't think I have write-access to this repo, so never mind the parenthesized comment above.

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+* Technical direction.
+* Project governance and process.
+* GitHub repository hosting.
+* Conduct guidelines.
+* Maintaining the list of additional Collaborators.
+* Development process and any coding standards.
+* Mediating technical conflicts between Collaborators.
+
+
+Not withstanding the above, the Projects and the entire technical community will follow any processes
```
Comment (Trott) [inline proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]:
```suggestion
Notwithstanding the above, the Projects and the entire technical community will follow any processes
```

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+as may be specified by the Board relating to the intake and license compliance review of contributions,
+including the OpenJS Foundation IP Policy.
+
+## Section 7. Elections
+
+Leadership roles in OpenJS Foundation will be peer elected
```
Comment (Trott) [inline proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]:
```suggestion
Leadership roles in OpenJS Foundation will be peer-elected
```

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md [hunk 10]
```diff
@@ -0,0 +1,320 @@
...
+For the first year after the formation of the OpenJS Foundation, the
+processes for filling this seat will be as follows:
+
+* As a constellation member of the OpenJS Foundation, the Node.js project
+  will elect one Board representative from either the Technical Steering Committee (TSC) or
+  Community Committee (CommComm). The representative will be selected according to a process
+  defined by the TSC and CommComm. 
+
+As described in the [Elections][] section a multiple candidate method
+will be be used for the voting for the first and second board members.
```
Comment (Trott) [inline proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]:
```suggestion
will be used for the voting for the first and second board members.
```

**Comment by Trott** [comment]
The document switches back and forth between "The OpenJS Foundation" and "OpenJS Foundation". Pick one and stick with it. (I prefer "The OpenJS Foundation" but either is fine.)

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md [hunk 11]
```diff
@@ -0,0 +1,320 @@
...
+OpenJS Foundation to propose changes using simple majority of the
+full OpenJS Foundation, the proposed changes being subject to review
+and approval by the Board. The Board may additionally make amendments to
+the OpenJS Foundation charter at any time, though the Board will not interfere
+with day-to-day discussions, votes or meetings
+of the CPC.
+
+## Section 3. Board’s Role in Setting OpenJS Foundation’s Strategic Direction. 
+
+The Board will set the overall CPC Policy. The policy will describe the
```
Comment (Trott) [inline proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]:
```suggestion
The Board will set the CPC Policy. The policy will describe the
```

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md [hunk 12]
```diff
@@ -0,0 +1,320 @@
...
+overarching scope of the OpenJS Foundation initiative, OpenJS Foundation 
+Foundation’s technical vision and direction and project release
+expectations in the form of expected cadence and intent. The Board will
+use the CPC as a delegate body for governing technical implementation,
+individual project scope and direction while they remain within the scope
+and direction of the policies as described in the CPC Policy document and
+approved by the Board.
+
+The Board sets the overall CPC policy through [Foundation mission and vision
```
Comment (Trott) [inline proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]:
```suggestion
The Board sets the CPC policy through [Foundation mission and vision
```

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md [hunk 13]
```diff
@@ -0,0 +1,320 @@
...
+
+* Observers
+* Regular members
+* Voting members
+
+
+### Observers
+
+Observers are free to attend meetings and participate in the work of the
+CPC as well as the consensus seeking process.
```
Comment (Trott) [inline proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]: This grants Observers the ability to block things that have consensus among CPC members. If that's intentional, cool. But if not, uh, you'll want to change that.

**Comment by Trott** [comment]
Usage of "shall" in this doc seems stilted and also a bit unclear. (Is it prescriptive--the CPC must do these things--or aspirational--the CPC should strive to do these things?) Switch to "must" or "will" as appropriate.

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md [hunk 14]
```diff
@@ -0,0 +1,320 @@
+# Cross Project Council (CPC) Charter
+
+## Section 1. Guiding Principle.
+
+The OpenJS Foundation will operate transparently, openly,
+collaboratively, and ethically. Project proposals, timelines, and status
+must not merely be open, but also easily visible to outsiders.
```
Comment (Trott) [inline proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]: Interpreted in a straightforward way, this sentence is meaningless. The phrase "easily visible to outsiders" is supposed to be in contrast to "open". Without further explanation, it's not clear what the distinction is supposed to be.

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md [hunk 15]
```diff
@@ -0,0 +1,320 @@
+# Cross Project Council (CPC) Charter
+
+## Section 1. Guiding Principle.
+
+The OpenJS Foundation will operate transparently, openly,
+collaboratively, and ethically. Project proposals, timelines, and status
```
Comment (Trott) [inline proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]:
```suggestion
collaboratively, and ethically. Project proposals, timelines, and statuses
```

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md (hunk 14 above)
Comment (Trott) [reply proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]: Easy possible solutions would be to remove the sentence or to simply state: "Project proposals, timelines, and statuses must be open." Or: "Project proposals, timelines, and statuses must be easily visible." Or "easily visible to outsiders." Or "easily visible to anyone." Lots of options.

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md (hunk 13 above)
Comment (MylesBorins) [reply proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]: That is intentional.

**Comment by MylesBorins** [comment]
@Trott I've sent you an invite so you can "go to town" on errata fixes and push some commits. I've gone ahead and landed all non-substantive changes you suggested so far and rebased them into a single additional commit.

For more substantial text changes we should likely go through another PR or at least bring it at next weeks bootstrap meeting. As the board meeting is on Friday I think it might make sense to punt those changes for now and do them as a follow up. Thoughts?

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md (hunk 1 above)
Comment (MylesBorins) [reply proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]: I think we should change that in a follow up PR

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md (hunk 14 above)
Comment (MylesBorins) [reply proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]: I think we should expand on this in a follow up PR

**Comment by Trott** [comment]
> For more substantial text changes we should likely go through another PR or at least bring it at next weeks bootstrap meeting. As the board meeting is on Friday I think it might make sense to punt those changes for now and do them as a follow up. Thoughts?

Sure.

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md [hunk 16]
```diff
@@ -0,0 +1,320 @@
...
+Responsibility for directing individual meetings may be
+delegated by the CPC Chairperson to any other CPC member. The CPC chairperson
+shall be one of the Voting members, and will be selected by the CPC members
+through consensus or if necessary a vote as described in the section titled "Voting".
+Minutes or an appropriate recording shall be taken and made available to
+the community through accessible public postings.
+
+Voting members have the roles and responsibilities as outlined below.
+
+Section 5. Responsibilities and Expectations of the CPC
```
Comment (mcollina) [inline proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]: This is missing some markdown headers

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md [hunk 17]
```diff
@@ -0,0 +1,320 @@
...
+   1. Identifying, recruiting and engaging prospective projects.
+   1. Voting members are responsible for making a final decision if consensus
+      cannot be reached among members.
+   1. Members of the CPC, or committees consisting of subsets of the CPC
+      membership will be expected to meet on a regular basis to discuss topics
+      such as new project acceptance into the mentorship program, project graduation
+      from the mentorship program, Project issues and conflicts, opportunities
+      for collaboration between Projects, opportunities for the Foundation in the
+      greater JavaScript community, etc.
+
```
Comment (mcollina) [inline proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]: I think we can make this paragraph simpler by splitting the responsibilities of voting members to the ones of regular members in two lists.

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md (hunk 14 above)
Comment (mhdawson) [reply proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]: I believe this text came from an earlier doc (not sure if TSC governance, something from the JSF or possibly Kubernetes) so I agree we should expand in a follow on PR.

Comment (MylesBorins) [reply proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]: Not going to include this in these changes. Lets do a follow up

File: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md (hunk 17 above)
Comment (MylesBorins) [reply proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md]: lets do this in a follow up

**Comment by MylesBorins** [comment]
approved by board today. Lets open new PRs for additional editorial changes

**Pull request merged**